### PR TITLE
Normalization testing; small `Activation` improvement

### PR DESCRIPTION
**Pull request by Linux-cpp-lisp**
## Description
Done:

- Avoid `torch.cat` when unneeded in `Activation`
- More tests for `Activation`
- Added `assert_normalized` and use for various modules
- Weight support in `assert_normalized`
- `assert_normalized` for `Linear`, `TensorProduct` tests

## How Has This Been Tested?
e3nn tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds or improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation improvement (updates to user guides, docstrings, or developer docs)

## Checklist:

- [X] I have read the [**CONTRIBUTING**](https://github.com/e3nn/e3nn/blob/main/CONTRIBUTING.md) document.
- [X] My code follows the code style of this project.
- [X] I have updated the documentation (if relevant).
- [X] I have added tests that cover my changes (if relevant).
- [X] All new and existing tests passed.
- [X] I have updated the [Changelog](https://github.com/e3nn/e3nn/blob/main/ChangeLog.md).

### CODE REVIEW

File: docs/guide/normalization.rst [hunk 1]
```diff
@@ -84,7 +84,7 @@ Testing
 
 You can use ``e3nn.util.test.assert_normalized`` to check whether a function or module is normalized at initialization:
 
-.. code::
+.. jupyter-execute::
```
Comment (Linux-cpp-lisp) [inline docs/guide/normalization.rst:87]: @mariogeiger I didn't do this since the cell has no output, and since I didn't want to run expensive (relatively) computations on readthedocs

Comment (Linux-cpp-lisp) [reply docs/guide/normalization.rst:87]: (since converging the stats for `Linear` takes larger settings than the default — it fails right now on the docs)

Comment (mariogeiger) [reply docs/guide/normalization.rst:87]: Ah ok, let me remove my commit then
